### PR TITLE
VS2017 Getting Started doc updates

### LINKED ISSUE #1480: Review topics that use Visual Studio 2017 RC
Some screenshots and instructions might have changed with the RC3 release.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #1480

(Attempting a squashed commits version here.)

Notes
* *Still rough! Ruf! Ruf!* :dog2: Even after review and merge, I'd like to come back in a week or two and work these a bit more.
* *New images!* ... but more needed in a few spots. Let's not hold it up tho just for a couple that I can add later.
* Propose we keep the old images in the media folder. They might be linked into other docs. After all docs in this area have dedicated static asset folders for their images, we can delete all of the images in the media folder safely.
* If there are more docs that need updating under #1480, I can do those, but I suppose not holding up the whole 🌏 planet 🌍 for these would be good.
* @mairaw I don't think you added the red stroke outlines to the images you did for #1663. I got it covered ... I opened #1675 for it.

I'll remove the `WIP` label after the build runs and I fix any broken links.

I'll ping when it's ready for a :eyes: look :eyes:.

### CODE REVIEW

File: docs/csharp/getting-started/debugging-with-visual-studio-2017.md [hunk 1]
```diff
@@ -1,143 +1,139 @@
...
+1. Program execution stops when it reaches the breakpoint and before the `Console.WriteLine` method executes. The **Autos** window displays the values of variables that are used around the current line. The **Locals** window displays the values of variables that are defined in the currently executing method.
 
-   ![Image](./media/breakpoint_2017.jpg)
+   ![Visual Studio application window](./media/debugging-with-visual-studio-2017/break.png)
 
-1. Let's try to change the value of the variables to see how this affects our program. If the **Immediate window** is not visible, display it by choosing the **Debug**, **Windows**, **Immediate** menu item. The **Immediate window** lets you interact with the application you're debugging.
+1. You can change the value of the variables to see how it affects your program. If the **Immediate Window** is not visible, display it by choosing the **Debug > Windows > Immediate** menu item. The **Immediate Window** lets you interact with the application you're debugging.
```
Comment (GuardRex) [inline docs/csharp/getting-started/debugging-with-visual-studio-2017.md]: *choosing* :point_right: *selecting*

File: .openpublishing.redirection.json [hunk 2]
```diff
@@ -242,6 +242,26 @@
...
         "source_path": "docs/tutorials/index.md", 
         "redirect_url": "/dotnet/articles/samples-and-tutorials/"
+    },
+    {
+        "source_path": "docs/csharp/getting-started/consuming-library-with-visual-studio-2017.md", 
+        "redirect_url": "/dotnet/articles/getting-started/consuming-library-with-visual-studio",
+        "redirect_document_id": true
+    },
+    {
+        "source_path": "docs/csharp/getting-started/debugging-with-visual-studio-2017.md",
```
Comment (GuardRex) [inline .openpublishing.redirection.json]: remove this redirect .... this one had to stay the same

File: .openpublishing.redirection.json [hunk 3]
```diff
@@ -242,6 +242,26 @@
...
+        "redirect_url": "/dotnet/articles/getting-started/debugging-with-visual-studio",
+        "redirect_document_id": true
+    },
+    {
+        "source_path": "docs/csharp/getting-started/library-with-visual-studio-2017.md", 
+        "redirect_url": "/dotnet/articles/getting-started/library-with-visual-studio",
+        "redirect_document_id": true
+    },
+    {
+        "source_path": "docs/csharp/getting-started/publishing-with-visual-studio-2017.md",
```
Comment (GuardRex) [inline .openpublishing.redirection.json]: remove this redirect .... this one had to stay the same

File: .openpublishing.redirection.json (hunk 2 above)
Comment (mairaw) [inline .openpublishing.redirection.json]: why?

Comment (GuardRex) [inline .openpublishing.redirection.json]: I couldn't rename that one (or the publishing one). The `debugging-with-visual-studio.md` and the `publishing-with-visual-studio.md` are 301 redirecting.

Did you want to ignore that and move the content over to those files anyway?

Comment (GuardRex) [inline .openpublishing.redirection.json]: Double-301? ... If a proxy or a system has a 301 on the non-2017 address and they go to the 2017-version of the doc ... it 301's back to the non-2017 doc. Hum. 🤔

**Review by mairaw — COMMENTED** [review]
I haven't finished my review, but so far, so good!

File: docs/csharp/getting-started/library-with-visual-studio.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+In this topic, you'll create a simple utility library that contains a single string-handling method. You'll implement it as an [extension method](../../csharp/programming-guide/classes-and-structs/extension-methods.md) so that you can call it as if it were a member of the @System.String class.
+
+## Creating a class library solution
+
+Start by creating a solution for your class library project and its related projects. A Visual Studio Solution just serves as a container for one or more projects. To create the solution:
+
+1. On the Visual Studio menu bar, choose **File > New > Project**.
+
+1. In the **New Project** dialog, expand the **Other Project Types** node, and select **Visual Studio Solutions**. Name the solution "ClassLibraryProjects" and select the **OK** button.
```
Comment (GuardRex) [inline docs/csharp/getting-started/library-with-visual-studio.md]: I was trying to follow the existing pattern where if the user is directed to type something the text is in quotes. In other places, the project name is in-line code.

Comment (mairaw) [inline docs/csharp/getting-started/library-with-visual-studio.md]: That's a good point.

**Comment by BillWagner** [comment]
@GuardRex Can you update the merge conflicts here as well?

**Comment by GuardRex** [comment]
@BillWagner Conflicts seem* to be fixed.

seem* = I better go back over the changes one more time and see how the report looks after the build finishes. I'll remove the WIP label and ping you when ready.

**Comment by GuardRex** [comment]
@BillWagner ready for :eyes:

**Review by BillWagner — APPROVED** [review]
Looks good overall. I'd agree with @mairaw that the Prev and Next buttons would be useful for readers.  Adding them would be a good idea, but not a blocker to merging.

**Pull request merged**